project.mml

```diff
@@ -267,7 +267,7 @@ Layer:
               OR "natural" IN ('water', 'glacier'))
             AND building IS NULL
             AND way_area > 0.01*!pixel_width!::real*!pixel_height!::real
-          ORDER BY z_order, way_area DESC
+          ORDER BY COALESCE(layer,0), way_area DESC
```

Comment:
This ends up being equivalent because the C transform z_order goes first by layer, and unless a water area happens to have a road tag, it's the same.